src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -247,6 +250,7 @@ private static string ComputeGuid(MetadataReader mr)
             int methodToken = MetadataTokens.GetToken(mr, mdefh);
             MethodDefinition mdef = mr.GetMethodDefinition(mdefh);
             int rva = mdef.RelativeVirtualAddress;
+            string methodName = mr.GetString(mdef.Name);
```

Comment:
this isn't the full name of the method (ie: `NameSpace.ClassName+NestedClassName<T,U>.SomeMethod<V,W>(Arg1 arg1, Arg2 arg2, ..., ArgN argN)`). It's just `SomeMethod`.  Is that what we want?  The file might have 100 methods all named `ToString` or something.

Reply:
You are right. Do you know how to get the method full signature?

Reply:
There's not an easy way :-(

you need to get the `mdef.GetDeclaringType()` which will be a `TypeDefinitionHandle`.  then you need to get the `TypeDefinition` using `mr.GetTypeDefiniton()` which will have a `Namespace` and a `Name` property.  And also you have to get its `GetDeclaringType` if it's a nested type, and so on until there's no more enclosing types.

That will at least get you enough so you can print `SomeNameSpace.SomeClass.SomeEnclosingType.SomeMethodName` which is better.  (it woudl be good to also print the method signature in case we need to care about overloads, but that's harder)